internal/dag/httpproxy_processor.go

```diff
@@ -228,7 +228,13 @@ func (p *HTTPProxyProcessor) computeHTTPProxy(proxy *contour_api_v1.HTTPProxy) {
 				return
 			}
 
-			svhost := p.dag.EnsureSecureVirtualHost(HTTPS_LISTENER_NAME, host)
+			listener, err := p.dag.GetSingleListener("HTTPS")
```

Comment:
Note to self: we talked about using a custom Listener protocol for this case, e.g. `projectcontour.io/https` or something along those lines, to indicate the TLS details will be contained in an HTTPProxy or Ingress rather than the Listener itself. This is probably more intuitive than using protocol = HTTPS which then requires mode: passthrough in order to not specify certificate details in the Listener. ref. https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1beta1/gateway_types.go#L247-L251

Reply:
related to migration/integration I had and wanted to maybe spike on seeing how complicated it would be to be able to reference HTTPProxies/Ingresses from a Gateway Listener, could be made to be compatible with the method for specifying a custom protocol you mention in this thread and a migration step for users as they move to Gateway API

Reply:
would make integrating with multiple listener ports a little easier in theory

Reply:
OK, with https://github.com/projectcontour/contour/pull/5160/commits/2b7f149edf9239f43c1bff64a37ad1a0f8a32ebc, the following Gateway spec can be used with HTTPProxies/Ingresses that terminate TLS:

```yaml
kind: Gateway
apiVersion: gateway.networking.k8s.io/v1beta1
metadata:
  name: contour
  namespace: projectcontour
spec:
  gatewayClassName: contour
  listeners:
    - name: http
      port: 80
      protocol: HTTP
      allowedRoutes:
        namespaces:
          from: All
    - name: https
      port: 443
      protocol: projectcontour.io/https # using this protocol means `tls` does not need to be specified for this Listener
      allowedRoutes:
        namespaces:
          from: All
```